readthedocs/subscriptions/managers.py

```diff
@@ -130,31 +130,22 @@ def update_from_stripe(self, *, rtd_subscription, stripe_subscription):
                 log.error(
                     'Stripe subscription trial end date invalid. ',
                     trial_end_date=trial_end_date,
-                    stripe_subscription=stripe_subscription.id,
                 )
 
         # Update the plan in case it was changed from the Portal.
-        # Try our best to match a plan that is not custom. This mostly just
-        # updates the UI now that we're using the Stripe Portal. A miss here
-        # just won't update the UI, but this shouldn't happen for most users.
-        from readthedocs.subscriptions.models import Plan
-        try:
-            plan = (
-                Plan.objects
-                # Exclude "custom" here, as we historically reused Stripe plan
-                # id for custom plans. We don't have a better attribute to
-                # filter on here.
-                .exclude(slug__contains='custom')
-                .exclude(name__icontains='Custom')
-                .get(stripe_id=stripe_subscription.plan.id)
-            )
-            rtd_subscription.plan = plan
-        except (Plan.DoesNotExist, Plan.MultipleObjectsReturned):
-            log.error(
-                'Plan lookup failed, skipping plan update.',
-                stripe_subscription=stripe_subscription.id,
-                stripe_plan=stripe_subscription.plan.id,
-            )
+        # This mostly just updates the UI now that we're using the Stripe Portal.
+        # A miss here just won't update the UI, but this shouldn't happen for most users.
+        # NOTE: Previously we were using stripe_subscription.plan,
+        # but that attribute is deprecated, and it's null if the subscription has more than
+        # one item/plan, we have a couple of subscriptions that have more than
+        # one item, so we use the first that is found in our DB.
+        for stripe_item in stripe_subscription["items"].data:
```

Comment:
Isn't this `stripe_subscription.items.data` instead? I haven't seen accessing Stripe objects as dictionaries before.

Reply:
Also, I haven't found that `stripe_item` has a `.plan` attribute either 🤔

Reply:
>  Isn't this stripe_subscription.items.data instead? I haven't seen accessing Stripe objects as dictionaries before.

The stripe object inherits from a dictionary, using `.items` will access the `.items()` method instead of accessing the data.

> Also, I haven't found that stripe_item has a .plan attribute either

Got that one from the webhook response, but looks like the whole Plan object is deprecated.

> You can now model subscriptions more flexibly using the [Prices API](https://stripe.com/docs/api/plans?lang=python#prices). It replaces the Plans API and is backwards compatible to simplify your migration.

And checking, looks like it's backwards compatible, the price id is the same as the plan id. Changing